src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx

```diff
@@ -153,7 +153,7 @@ export const Profile_PrivacyScreen = () => {
                   ),
                   destructiveArrowFunction: async () => {
                     setIsCleaningCollectedData(true);
-                    await deleteCollectedData();
```

Comment:
Sure that `setIsCleaningCollectedData(false);` should be called immediately without waiting for the actual deletion to complete?

Reply:
TS complains if I do that, internally to await is handled, so I guess is safe, do you get that warning?

<img width="545" alt="image" src="https://github.com/AtB-AS/mittatb-app/assets/43450882/26aa9da0-122a-4522-9a4b-fd12cac8f521">

Reply:
Hmm, in `BeaconsContextState`, I think you can set the type to `deleteCollectedData: () => Promise<void>;`

Reply:
Ok, I made some fixes to it. to allow void and boolean promises